### ISSUE TITLE
"is in the list" condition always evaluates to "false" for numerical attributes

### ISSUE THREAD

**Issue by oleksandr-bezushko-axon** [growthbook/growthbook-kotlin#65]
### Steps to reproduce
1. Have a numerical attribute in the map supplied to `GBSDKBuilder(attributes = attributesMap)`, for example app version code:
```kotlin
val versionCode: Int = TODO()
val attributesMap = mapOf(
    "appBuildNumber" to versionCode,
    // Other attributes...
)
```
2. In Growthbook console, add a "forced override rule" for a feature. The override rule should have a condition "is in the list" for that numerical attribute:
![image](https://github.com/growthbook/growthbook-kotlin/assets/75365247/6c273298-188b-4672-8408-c20762954ca9)
3. Use an app that has versionCode equal to be one of the values in the list (like 3440)

### Expected
The override rule gets applied to my app, and the feature's value gets overridden.

### Actual
The override rule is not applied.

### Environment
Growthbook SDK version: 1.1.34
Growthbook's CDN

### Additional information
Override rules work if I create separate rules for each appBuildNumber with "is equal to" condition:
![image](https://github.com/growthbook/growthbook-kotlin/assets/75365247/c35277e2-52be-449a-a708-1fa0c2505c94)

----------------

It seems that Growthbook SDK converts all attributes to strings while evaluating a condition:
![image](https://github.com/growthbook/growthbook-kotlin/assets/75365247/726c66b9-ca33-4498-ba06-2360109ccd6a)

But the condition JSON that comes from your server provides an array of numbers, causing the following "contains" to fail because of type mismatch (array of numbers doesn't contain a string):
![image](https://github.com/growthbook/growthbook-kotlin/assets/75365247/dc2dc41e-1546-4303-b1a2-b096e801d823)

**Comment by Bohdan-Kim** [comment]
Hi Oleksandr Bezushko!

Thanks for reporting such a bug. We've reproduced it and will be working on it.
I also noticed that when the values list in GowthBook console inputted as strings the Forced rule is applied.


![image](https://github.com/growthbook/growthbook-kotlin/assets/135710078/0213ec2e-e886-4e65-95d8-46a3927e333c)
![image](https://github.com/growthbook/growthbook-kotlin/assets/135710078/8d52bd43-084e-45df-b816-208522f0f5b6)

**Comment by Bohdan-Kim** [comment]
[Pull request](https://github.com/growthbook/growthbook-kotlin/pull/68) was created according to this issue

**Comment by andrew-k-21-12** [comment]
Hello,

It is still getting reproduced for me when using a combination of `io.growthbook.sdk:GrowthBook:5.0.0` in Android and `Build: 3.6.0+6559b64 (2025-05-01)` for the server. Even more, in this case `is equal to` conditions in forced rules stop to work as well.

To make it work, I do 2 things:
1. Send all `Number` attributes as `GBString` instead of `GBNumber` from Android.
2. For `is in the list` conditions it's needed to turn **Target by Attributes** into **Advanced mode** and wrap all list values into string quotes (`"`) as mentioned [here](https://github.com/growthbook/growthbook-kotlin/issues/65#issuecomment-1685870569) above:

<img width="781" alt="Image" src="https://github.com/user-attachments/assets/0e00798f-6428-41d8-a1a7-45ab1c8bb4d8" />

**Comment by andrew-k-21-12** [comment]
If the test passes, then everything is fine. 😅 

But for some reason in the server response I receive condition for a number attribute in a string format:

```json
"condition":{"userId":"123"}
```

This is why #1 from [this comment](https://github.com/growthbook/growthbook-kotlin/issues/65#issuecomment-2912716889) works, I guess.

**Comment by Bohdan-Kim** [comment]
Is it reproduced on [app.growthbok.io](url)?

**Comment by andrew-k-21-12** [comment]
It is reproduced on self-host. Debugging it right now.

**Comment by andrew-k-21-12** [comment]
Okay, I've found the reason for the issue. It happens in Android SDK in `com.sdk.growthbook.evaluators.GBConditionEvaluator#evalConditionValue`. This comparison returns `false` while it should return `true`. For some reason 2 `GBNumber`s with the same underlying value are not equal (I've erased the actual values but they are equal):

<img width="681" alt="Image" src="https://github.com/user-attachments/assets/b81da861-0a35-4def-85bd-da2fd03abd3a" />

Will try to provide a short unit test demonstrating this issue.
